resources/js/pack-it-forms.js

```diff
@@ -946,6 +946,14 @@ function emptystr_if_null(argument) {
     return argument ? argument : "";
 }
 
+/* Ping the server periodically, to let it know this page has not been closed. */
+function ping() {
+    var img = new Image();
+    img.src = '/ping';
+    img = undefined;
+}
+setInterval(ping, 30000); // call ping every 30 seconds
+
 /* --- Cross-browser convenience functions --- */
```

Comment:
Since I'd like to be able to support multiple uses of these forms, I think the best thing to do here is to make the ping functionality (including whether it is enabled) controlled by data provided to the form by the integration as well.

Ah, I see that you did this in a later commit as well. It looks good too.